CRM/Lineitemedit/Util.php

```diff
@@ -988,14 +988,31 @@ public static function generatePriceField() {
     $priceFieldValueParams = $priceFieldValue;
     unset($priceFieldValueParams['id'], $priceFieldValueParams['name'], $priceFieldValueParams['weight']);
     for ($i = 1; $i <= 10; ++$i) {
-      $p = civicrm_api3('PriceField', 'create', array_merge($priceFieldParams, ['label' => ts('Additional Line Item') . " $i"]));
-      civicrm_api3('PriceFieldValue', 'create', array_merge(
-        $priceFieldValueParams,
-        [
-          'label' => ts('Additional Item') . " $i",
-          'price_field_id' => $p['id'],
-        ]
-      ));
+      $params = array_merge($priceFieldParams, ['label' => ts('Additional Line Item') . " $i"]);
+      $priceField = civicrm_api3('PriceField', 'get', $params)['values'];
+      if (empty($priceField)) {
+        $p = civicrm_api3('PriceField', 'create', $params);
+        civicrm_api3('PriceFieldValue', 'create', array_merge(
+          $priceFieldValueParams,
+          [
+            'label' => ts('Additional Item') . " $i",
+            'price_field_id' => $p['id'],
+          ]
+        ));
+      }
+      else {
+        civicrm_api3('PriceField', 'create', ['id' => key($priceField), 'is_active' => TRUE]);
+      }
+    }
+  }
+
+  public static function disableEnablePriceField($enable = FALSE) {
```

Comment:
Could you improve the name here? Do you mean disablePriceField() ? Or better: disableLineItemEditPriceFields() ?